pluginframework/common/go_to_tf.go

```diff
@@ -0,0 +1,222 @@
+package pluginframework
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/databricks/databricks-sdk-go/logger"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/databricks/terraform-provider-databricks/internal/reflect_utils"
+)
+
+// Converts a gosdk struct into a tfsdk struct, with the folowing rules.
+//
+//	string -> types.String
+//	bool -> types.Bool
+//	int64 -> types.Int64
+//	float64 -> types.Float64
+//	string -> types.String
+//
+// NOTE:
+//
+//	If field name doesn't show up in ForceSendFields and the field is zero value, we set the null value on the tfsdk.
+//	types.list and types.map are not supported
+//	map keys should always be a string
+//	tfsdk structs use types.String for all enum values
+//	non-json fields will be omitted
+func GoSdkToTfSdkStruct(gosdk interface{}, tfsdk interface{}, ctx context.Context) diag.Diagnostics {
+
+	srcVal := reflect.ValueOf(gosdk)
+	destVal := reflect.ValueOf(tfsdk)
+
+	if srcVal.Kind() == reflect.Ptr {
+		srcVal = srcVal.Elem()
+	}
+
+	if destVal.Kind() != reflect.Ptr {
+		return diag.Diagnostics{diag.NewErrorDiagnostic("please provide a pointer for the tfsdk struct", "tfsdk to gosdk struct conversion failure")}
+	}
+	destVal = destVal.Elem()
+
+	if srcVal.Kind() != reflect.Struct || destVal.Kind() != reflect.Struct {
+		return diag.Diagnostics{diag.NewErrorDiagnostic(fmt.Sprintf("input should be structs %s, %s", srcVal.Type().Name(), destVal.Type().Name()), "tfsdk to gosdk struct conversion failure")}
+	}
+
+	var forceSendFieldVal []string
+
+	forceSendField := srcVal.FieldByName("ForceSendFields")
+	if !forceSendField.IsValid() {
+		// If no forceSendField, just use an empty list.
+		forceSendFieldVal = []string{}
+	} else {
+		forceSendFieldVal = forceSendField.Interface().([]string)
+	}
+
+	for _, field := range reflect_utils.ListAllFields(srcVal) {
+		srcField := field.V
+		srcFieldName := field.Sf.Name
+
+		srcFieldTag := field.Sf.Tag.Get("json")
+		if srcFieldTag == "-" {
+			continue
+		}
+		destField := destVal.FieldByName(srcFieldName)
+
+		if !destField.IsValid() {
+			logger.Tracef(ctx, fmt.Sprintf("field skipped in gosdk to tfsdk conversion: destination struct does not have field %s", srcFieldName))
+			continue
+		}
+
+		err := goSdkToTfSdkSingleField(srcField, destField, fieldInForceSendFields(srcFieldName, forceSendFieldVal), ctx)
+		if err != nil {
+			return diag.Diagnostics{diag.NewErrorDiagnostic(err.Error(), "gosdk to tfsdk field conversion failure")}
+		}
+	}
+	return nil
+}
+
+func goSdkToTfSdkSingleField(srcField reflect.Value, destField reflect.Value, forceSendField bool, ctx context.Context) error {
+
+	if !destField.CanSet() {
+		panic(fmt.Errorf("destination field can not be set: %s", destField.Type().Name()))
```

Comment:
Anywhere that we panic, we should print an error that says something like "This is a bug. Please report this to the maintainers at github.com/databricks/terraform-provider-databricks."